firefox_puppeteer/ui/tabbar.py

```diff
@@ -3,126 +3,315 @@
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
 from marionette import (
-    HTMLElement,
     Wait,
 )
 
-from marionette.errors import NoSuchElementException, StaleElementException
+from marionette.errors import NoSuchElementException
+
+import firefox_puppeteer.errors as errors
 
 from .. import DOMElement
-from ..base import BaseLib
+from ..base import UIBaseLib
 
 
-class Tabs(BaseLib):
+class TabBar(UIBaseLib):
 
-    # TODO arrow scrollers, drop down list
+    # Properties for visual elements of the tabs toolbar #
 
     @property
-    def newtab_button(self):
-        """
-        :returns: The new tab button element.
+    def menupanel(self):
+        """A :class:`~ui.menu.MenuPanel` instance which represents the menu panel
+        at the far right side of the tabs toolbar.
+
+        :returns: :class:`~ui.menu.MenuPanel` instance
         """
-        return (self.marionette.find_element('id', 'tabbrowser-tabs')
-                               .find_element('anon attribute',
-                                             {'anonid': 'tabs-newtab-button'}))
+        return MenuPanel(lambda: self.marionette, self.window)
 
     @property
-    def active_tab(self):
+    def newtab_button(self):
```

Comment:
We have blocks here to make it easier to find them:

1. Properties for ui elements
2. Properties for helpers
3. Helper methods

I think that this is pretty helpful in case of large classes when you would have to scroll a lot up and down.